### ISSUE TITLE
Python 3.7 failures on 32bit systems

### ISSUE THREAD

**Issue by hroncok** [MagicStack/uvloop#172]
* **uvloop version**: master
* **Python version**: 3.7.0rc1
* **Platform**: Linux (Fedora 29), 32bit only (both i383 and armv7hl)
* **Can you reproduce the bug with `PYTHONASYNCIODEBUG` in env?**: yes

The test suite fails on Python 3.7 on 32bit Linux.

It's a couple of errors (256) but it look like all are related to:

```python
Traceback (most recent call last):
  File "/builddir/uvloop/uvloop/_testbase.py", line 81, in setUp
    self.loop = self.new_loop()
  File "/builddir/uvloop/uvloop/_testbase.py", line 299, in new_loop
    return uvloop.new_event_loop()
  File "/builddir/uvloop/uvloop/__init__.py", line 20, in new_event_loop
    return Loop()
  File "uvloop/loop.pyx", line 110, in uvloop.loop.Loop.__cinit__
    self._transports = weakref_WeakValueDictionary()
TypeError: 'NoneType' object is not callable
```

This might be related to reference counting and the fact that weakref is now not loaded at start time:

```pycon
Python 3.6.5 (default, Mar 29 2018, 18:20:46) 
[GCC 8.0.1 20180317 (Red Hat 8.0.1-0.19)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> 'weakref' in sys.modules
True
```

```pycon
Python 3.7.0rc1 (default, Jun 12 2018, 11:19:29) 
[GCC 8.1.1 20180502 (Red Hat 8.1.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> 'weakref' in sys.modules
False
```

Why this happens only on 32bits, I have no idea.

**Comment by 1st1** [comment]
This is weird.  The `weakref` module is referenced from `uvloop/includes/__init__.py`, and the `includes` module is then imported from `uvloop/__init__.py` to specifically guard against this scenario.

Seems like a bug in Python 3.7 to be honest.

**Comment by jamesjer** [comment]
Actually, the first error I see in the logs is this one:
```
ERROR: test_aiohttp (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_aiohttp
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/loader.py", line 434, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.7/unittest/loader.py", line 375, in _get_module_from_name
    __import__(name)
  File "/builddir/build/BUILD/uvloop-0.10.1/tests/test_aiohttp.py", line 12, in <module>
    from uvloop import _testbase as tb
  File "/builddir/build/BUILD/uvloop-0.10.1/uvloop/__init__.py", line 7, in <module>
    from .loop import Loop as __BaseLoop  # NOQA
  File "uvloop/includes/stdlib.pxi", line 128, in init uvloop.loop
    cdef long MAIN_THREAD_ID = <long>threading.main_thread().ident
OverflowError: Python int too large to convert to C long
```

Changing the type from long to long long fixes that, and then the other errors all disappear.

**Comment by hroncok** [comment]
Oh. I should have posted full log in the first place. Nice catch!

**Comment by 1st1** [comment]
Should be fixed in master, please test. Thanks!

**Comment by vstinner** [comment]
It seems like the bug has been fixed by the commit a68f3c9a30a888d0bd28d608f6113f749e9c9cc9.

Most uvloop tests pass on my 32-bit Ubuntu and Python master (future 3.8). I have a few failures coming from the fact that I didn't install Python, but ran it from the directory where I compiled it from source.
